### PR TITLE
[risk=no][rw-4980] Move ProfileController business logic to ProfileService

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

This is a CRUDE, MECHANICAL refactoring to push basically all of the business logic from ProfileController to ProfileService.  Only a few minor updates here, noted by inline comments.  There are many rough edges here.  This is only one step of the process.  My goal is to separate these steps by PR in order to keep them manageable for quick review.

A few obvious improvements for these future PRs:
* Don't manipulate DbUser objects at the Controller level
* Don't return HTTP Response objects at the Service level

TODO: move tests

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 1]
```diff
@@ -428,134 +173,46 @@ private void checkUserCreationNonce(DbUser user, String nonce) {
...
   @Override
   @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
   public ResponseEntity<UserListResponse> getAllUsers() {
     UserListResponse response = new UserListResponse();
-    List<Profile> responseList = new ArrayList<>();
-    for (DbUser user : userDao.findUsers()) {
-      responseList.add(profileService.getProfile(user));
-    }
-    response.setProfileList(responseList);
+    response.setProfileList(profileService.listAllProfiles());
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java:208]: Effectively the same - perhaps more efficient?

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 2]
```diff
@@ -217,7 +92,7 @@ private DbUser initializeUserIfNeeded() {
     // the CDR); we will probably need a job that deactivates accounts after some period of
     // not accepting the terms of use.
 
-    DbUser dbUser = initializeUserIfNeeded();
+    DbUser dbUser = profileService.initializeUserIfNeeded();
     profileAuditor.fireLoginAction(dbUser);
     return getProfileResponse(dbUser);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java:97]: There's no real need to have a local DbUser object but this pattern is everywhere.  Deferring updating this to future PRs.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 3]
```diff
@@ -230,124 +105,24 @@ private DbUser initializeUserIfNeeded() {
...
-                } catch (MessagingException e) {
-                  throw new WorkbenchException(e);
-                }
-              });
-    }
-
-    // Note: Avoid getProfileResponse() here as this is not an authenticated request.
-    final Profile createdProfile = profileService.getProfile(user);
     profileAuditor.fireCreateAction(createdProfile);
+    // Note: Avoid getProfileResponse() here as this is not an authenticated request.
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java:111]: I don't actually understand this comment, so I'm not removing it

File: api/src/main/java/org/pmiops/workbench/profile/ProfileService.java [hunk 4]
```diff
@@ -337,4 +391,303 @@ private void validateStringLength(String field, String fieldName, int max, int m
   public List<Profile> listAllProfiles() {
     return userService.getAllUsers().stream().map(this::getProfile).collect(Collectors.toList());
   }
+
+  private void verifyCaptcha(String captchaToken) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:395]: everything here is moved as-is from ProfileController except where noted.  Please do NOT review this moved code except to note whether it has been moved correctly.

File: api/src/main/java/org/pmiops/workbench/profile/ProfileService.java [hunk 5]
```diff
@@ -337,4 +391,303 @@ private void validateStringLength(String field, String fieldName, int max, int m
...
+    }
+  }
+
+  private boolean userHasEverLoggedIn(
+      com.google.api.services.directory.model.User googleUser, DbUser user) {
+    return user.getFirstSignInTime() != null || !googleUser.getChangePasswordAtNextLogin();
+  }
+
+  // TODO: should not be aware of ResponseEntity at the Service level
+  private ResponseEntity<Void> resetPasswordAndSendWelcomeEmail(String username, DbUser user) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:580]: We should fix these in a future PR

Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:580]: I'd prefer you don't introduce ResponseEntity into the service level. It's a simple operation to return the inner value here and wrap it in a ResponseEntity at the top level.

File: api/src/test/java/org/pmiops/workbench/profile/ProfileServiceTest.java [hunk 6]
```diff
@@ -63,18 +71,29 @@
   DbInstitution dbInstitution;
 
   @TestConfiguration
-  @MockBean({FreeTierBillingService.class, UserService.class})
   @Import({
+    ActionAuditQueryServiceImpl.class,
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/profile/ProfileServiceTest.java:75]: I think we need to make a list of needed tests for these and other methods.

**Pull request closed without merging**